pilot/pkg/xds/internalgen.go

```diff
@@ -153,3 +162,52 @@ func (sg *InternalGen) Generate(proxy *model.Proxy, push *model.PushContext, w *
 	}
 	return res
 }
+
+func (sg *InternalGen) registerWorkload(con *Connection) {
+	if sg.Store == nil || !con.node.Metadata.AutoRegister {
```

Comment:
Should there be an explicit check here for the feature gate instead of checking that sg.Store is nil?